### ISSUE TITLE
AttributeError: module 'pylint.checkers.utils' has no attribute 'is_test_condition'

### ISSUE THREAD

**Issue by ahartikainen** [PyCQA/pylint#4097]
Hi, I noticed an error with the latest PyPI release (`pylint==2.6.1`)

[ArviZ CI (azure devops)](https://dev.azure.com/ArviZ/ArviZ/_build/results?buildId=3960&view=logs&jobId=2849c516-c5ba-5a2b-4c6e-ab4816e6761e&j=2849c516-c5ba-5a2b-4c6e-ab4816e6761e&t=1cb4bbdd-6306-5ba6-42a5-0a624961daf0)

Not sure what is going on, but apparently python can't access `is_test_condition`. Maybe there is some import order problem or something similar or our CI just behaving wrong.

### Steps to reproduce
1. pip install pylint
2. pylint .

### Current behavior

`AttributeError: module 'pylint.checkers.utils' has no attribute 'is_test_condition'`

<details><summary>Traceback message</summary>

```
/usr/bin/bash --noprofile --norc /home/vsts/work/_temp/0069574e-9b2e-4af7-a7d5-711353132a5f.sh
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pylint/__main__.py", line 18, in <module>
    pylint.run_pylint()
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pylint/__init__.py", line 22, in run_pylint
    PylintRun(sys.argv[1:])
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pylint/lint/run.py", line 349, in __init__
    linter.check(args)
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 862, in check
    self._check_files(
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 896, in _check_files
    self._check_file(get_ast, check_astroid_module, name, filepath, modname)
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 922, in _check_file
    check_astroid_module(ast_node)
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 1054, in check_astroid_module
    retval = self._check_astroid_module(
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pylint/lint/pylinter.py", line 1099, in _check_astroid_module
    walker.walk(ast_node)
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pylint/utils/ast_walker.py", line 75, in walk
    self.walk(child)
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pylint/utils/ast_walker.py", line 75, in walk
    self.walk(child)
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pylint/utils/ast_walker.py", line 75, in walk
    self.walk(child)
  [Previous line repeated 1 more time]
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pylint/utils/ast_walker.py", line 72, in walk
    callback(astroid)
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 1039, in visit_boolop
    self._check_simplifiable_condition(node)
  File "/opt/hostedtoolcache/Python/3.8.7/x64/lib/python3.8/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 1006, in _check_simplifiable_condition
    if not utils.is_test_condition(node):
AttributeError: module 'pylint.checkers.utils' has no attribute 'is_test_condition'
```

</details>

### pylint --version output

pylint==2.6.1

**Comment by Pierre-Sassoulas** [comment]
Thank you for the report, we're having some trouble following the release of astroid `2.5` The temporary fix is to use:
```
pylint==2.6.0
astroid < 2.5
```

**Comment by bnsh** [comment]
I have a few test cases:

```
def div6():
    return [num for num in sorted([1, 2, 3, 4]) if (num % 2 == 0) and (num % 3 == 0)]
```

```
def func(blah):
    assert "x" not in kwargs or kwargs["x"] is None
```

I'll make a gist/Docker instance in a few minutes.

**Comment by hippo91** [comment]
@Pierre-Sassoulas what makes you think the pb is due to `astroid 2.5`?
I made two virtual environments, one with `2.6.1` and the other with `2.6.2`. Both of them have identical astroid 2.4.2. 
However with 2.6.1 some problems arise with `pylint.checker.utils`. When i make a diff between test_pylint_2.6.1/lib/python3.9/site-packages/pylint and test_pylint_2.6.2/lib/python3.9/site-packages/pylint some files disappear.
I cannot find those change in the git history. I am a little bit stuck. Have you any idea? Is it possible the release process of 2.6.1 failed?

**Comment by Pierre-Sassoulas** [comment]
>Is it possible the release process of 2.6.1 failed?

Yes, I released 2.6.1 from the previous 2.6 branch, it failed, then I realized that 2.6 was not really 2.6 and rebased the 2.6 branch on the 2.6.0 tag, released as 2.6.2 and it worked. So 2.6.1 disappeared from the git history when it was rebased (also it crashed see #4096)

In the moment this issue it looked like a regression caused by 2.6.1, but I'm not sure it's fixed which is why this issue not closed.

**Comment by Pierre-Sassoulas** [comment]
I can reproduce with `2.6.1`, but this has been fixed in `2.6.2` and `master`. Spoiler, you'll have a few suggestions for performance in your tests by using generator in `any` and `all` when `2.7.0` is out thanks to @ikraduya 😉
